pybedtools/bedtool.py

```diff
@@ -3755,16 +3836,17 @@ def __init__(self, stream):
         self.pysam_bamfile = pysam.Samfile(self.stream)
 
     def _aligned_segment_to_interval(self, r):
+
         if r.rname >= 0:
-            rname = self.pysam_bamfile.getrname(r.rname)
+            rname = self.pysam_bamfile.getrname(r.rname) # TODO: getrname method is deprecated the up-to-date method is get_reference_name
         else:
             rname = "*"
 
         if r.rnext >= 0:
             if r.rnext == r.rname:
                 rnext = "="
             else:
-                rnext = self.pysam_bamfile.getrname(r.rnext)
+                rnext = self.pysam_bamfile.getrname(r.rnext) # TODO: getrname method is deprecated the up-to-date method is get_reference_name
```

Comment:
I have not looked into when the method was deprecated in pysam... but it should be changed if the version of mysam on the requirements already has the new method?